src/Entity/Task.php

```diff
@@ -128,6 +129,16 @@
  *         }
  *       }
  *     },
+ *     "task_add_to_group"={
+ *       "method"="PUT",
+ *       "path"="/tasks/{id}/add_to_group/{group_id}",
+ *       "controller"=AddToGroup::class,
+ *       "denormalization_context"={"groups"={"task_operation"}},
+ *       "access_control"="is_granted('ROLE_ADMIN') or is_granted('edit', object)",
```

Comment:
How does that work? Because if I remove `is_granted('ROLE_ADMIN')` and in the test replace `And the user "bob" has role "ROLE_ADMIN"` with `And the user "bob" has role "ROLE_USER"` the operation is executed without errors.

Also seems like there was an error in what I did here `is_granted('edit', object)` because if I put only that constraint doesn't matter what role user has, the operation always returns a 403.

Reply:
> How does that work?


When there is a `security` or `access_control` property, the [services tagged with `security.voter`](https://github.com/coopcycle/coopcycle-web/blob/master/app/config/services.yml#L1007-L1038) are invoked, and their `supports` method is called, to check if they are able to perform the necessary checks. 

Your problem here is that [`TaskOperationsVoter`](https://github.com/coopcycle/coopcycle-web/blob/9232b245c848668306dc473a359fa6f64fd64c31/src/Security/TaskOperationsVoter.php) only supports the `view` attribute. But if you change the operation to be more "restful" (see other comment), your operation definition would be on `TaskGroup`, and the associated voter supports the `edit` attribute.

Reply:
So, sorry, my comment was wrong ☹️